src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -43,14 +44,85 @@ public partial class GenericVectorTests
         private void TestConstructorWithSpan<T>() where T : struct
         {
             T[] values = GenerateRandomValuesForVector<T>().ToArray();
-            var valueSpan = new Span<T>(values);
+            Vector<T> vector;
 
-            var vector = new Vector<T>(valueSpan);
+            // Span<T> ctor
+            vector = new Vector<T>(new Span<T>(values));
             ValidateVector(vector,
                 (index, val) =>
                 {
                     Assert.Equal(values[index], val);
                 });
+
+            // ReadOnlySpan<T> ctor
+            vector = new Vector<T>(new ReadOnlySpan<T>(values));
+            ValidateVector(vector,
+                (index, val) =>
+                {
+                    Assert.Equal(values[index], val);
+                });
+
+            // ReadOnlySpan<byte> ctor
+            vector = new Vector<T>(MemoryMarshal.AsBytes(new ReadOnlySpan<T>(values)));
```

Comment:
Is this validation path particularly useful? It seems like if taking a `Span` works, it should work and things like `MemoryMarshal.AsBytes` returning a functioning Span should have been tested as part of `MemoryMarshal`?